core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -816,6 +816,10 @@ public boolean append(Object kbase, long koff, int klen, Object vbase, long voff
           } catch (SparkOutOfMemoryError oom) {
             canGrowArray = false;
           }
+        } else if (numKeys >= growthThreshold && longArray.size() / 2 >= MAX_CAPACITY) {
+          // The map has reached its growth threshold, but is already at MAX_CAPACITY and cannot
+          // grow. Instead, we prevent the map from accepting any more new elements.
```

Comment:
Can we be more clear in the comment, like:

```
... Instead, we prevent the map from accepting any more new elements to
make sure we don't exceed load factor, so we can reuse the array in sorting later.
```

Reply:
Done. I also changed the wording to reference UnsafeKVExternalSorter explicitly.

Reply:
Okay, thanks!